src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,366 @@
+        let mut sources = Vec::new();
+        if let Some(local_call) = local_call {
+            let method_pointer = &local_call.definition;
+            let expression_id  = &local_call.call;
+            let current_index = self.current_index.get();
+            let next_index = current_index + 1;
+
+            let breadcrumb_exists =
+                self.breadcrumbs.borrow_mut().get(current_index).contains_if(|breadcrumb| {
+                    breadcrumb.info.expression_id == *expression_id
+                });
+
+            if breadcrumb_exists {
+                debug!(self.logger, "Entering an existing {method_pointer.name} breadcrumb.");
+                //TODO[dg]: Highlight breadcrumb.
+            } else {
+                debug!(self.logger, "Creating a new {method_pointer.name} breadcrumb.");
+                self.remove_breadcrumbs_history(self.current_index.get());
+                let breadcrumb = Breadcrumb::new(&self.scene, method_pointer, expression_id);
+                let network = &breadcrumb.frp.network;
+                let breadcrumb_index = next_index;
+                let model = self.clone_ref();
+
+
+                // === User Interaction ===
+
+                frp::extend! { network
+                    eval_ breadcrumb.frp.outputs.selected(
+                        model.select_breadcrumb(breadcrumb_index).iter().for_each(|source| {
+                            source.emit(())
+                        })
+                    );
+                }
+
+                info!(self.logger, "Pushing {breadcrumb.info.method_pointer.name} breadcrumb.");
+                breadcrumb.set_position(Vector3(self.width(),0.0,0.0));
+                sources.push(breadcrumb.frp.fade_in.clone_ref());
+                self.breadcrumbs_container.add_child(&breadcrumb);
+                self.breadcrumbs.borrow_mut().push(breadcrumb);
+            }
+            self.current_index.set(next_index);
+            sources.append(&mut self.update_selection());
+        }
+        sources
+    }
+
+    fn pop_breadcrumb(&self) -> Vec<frp::Source> {
+        debug!(self.logger, "Popping {self.current_index.get()}");
+        if self.current_index.get() > 0 {
+            info!(self.logger, "Popping breadcrumb view.");
+            self.current_index.set(self.current_index.get() - 1);
+            self.update_selection()
+        } else {
+            default()
+        }
+    }
+
+    fn update_selection(&self) -> Vec<frp::Source> {
+        let mut sources = Vec::new();
+        let current_index = self.current_index.get();
+        for (index,breadcrumb) in self.breadcrumbs.borrow_mut().iter().enumerate() {
+            if index + 1 == current_index {
+                sources.push(breadcrumb.frp.select.clone_ref());
+            } else {
+                sources.push(breadcrumb.frp.deselect.clone_ref());
+            }
+        }
+        sources
+    }
```

Comment:
Just return two vectors - one with indexes of breadcrumbs to be deselected and the other with ids to be selected. Then in the frp network take these vectors, use the `<=` operator to iterate over the IDs, and for each id call something like `model.select_by_id(id)`.